common/djangoapps/student/tests/test_auto_auth.py

```diff
@@ -175,7 +175,40 @@ def test_json_response(self, course_id, course_key):  # pylint: disable=unused-a
             response_data
         )
 
-    def _auto_auth(self, params=None, **kwargs):
+    @ddt.data(*COURSE_IDS_DDT)
+    @ddt.unpack
+    def test_redirect_to_course(self, course_id, course_key):
+
+        # Create a user and enroll in a course
+        response = self._auto_auth({
+            'username': 'test',
+            'course_id': course_id,
+            'redirect': True,
+            'staff': 'true',
+        }, status_code=302)
+
+        # Check that a course enrollment was created for the user
+        self.assertEqual(CourseEnrollment.objects.count(), 1)
+        enrollment = CourseEnrollment.objects.get(course_id=course_key)
+        self.assertEqual(enrollment.user.username, "test")
+
+        # Check that the redirect was to the course info/outline page
+        urls = ('/info', 'course/{}'.format(course_key.to_deprecated_string()))
+        self.assertTrue(response.url.endswith(urls))  # pylint: disable=no-member
+
+    def test_redirect_to_main(self):
+        # Create user and redirect to 'home' (cms) or 'dashboard' (lms)
```

Comment:
It's because these are in common/djangoapps.  These tests are run with both `paver test_system -s lms` and `paver test_system -s cms`.  It can only test one of these conditions at a time, but the test doesn't know which context it is being run in.